websites/T/TOD/presence.ts

```diff
@@ -0,0 +1,281 @@
+class TOD extends Presence {
+	constructor(presenceOptions: PresenceOptions) {
+		super(presenceOptions);
+	}
+
+	getVideo() {
+		return document.querySelector<HTMLMediaElement>(
+			"video"
+		);
+	}
+
+	getVideoType() {
+		if (this.getVideoTitle().match(/S\d+\s*\|\s*E\d+/i) || this.getVideoTitle().match(/EP\s*\d+/i))
+			return "show";
+		 else 
+			return "movie";
+	}
+
+	getTitle(eyebrow = false) {
+		if (this.isWatching() && eyebrow) return this.getVideoTitle();
+
+		const title = document.querySelector("div.bein-dh1-hero__head-container > h1.bein-dh1-hero__title.heading-shadow")?.textContent;
+
+		return (
+			title ??
+			this.getVideoTitle().replace(/\d+\s*-\s*\d+/ as RegExp, "vs") ??
+			"Unknown"
+		);
+	}
+
+	getVideoTitle() {
+		return document.querySelector(
+			"div.diva-standard-title"
+		)?.textContent;
+	}
+
+	isWatching() {
+		return !!this.getVideoTitle();
+	}
+
+	isLive() {
+		if (document.querySelector("span.diva-live-now"))
+			return true;
+		else
+			return false;
+	}
+
+	isTrailer() {
+		return this.getVideoTitle().includes("Trailer") ? true : false;
+	}
+}
+
+const enum Assets {
+	Logo = "https://i.imgur.com/2euETKm.png",
+}
+
+const presence = new TOD({
+		clientId: "1229087342787039334",
+	}),
+	data: {
+		startedSince: number;
+		settings?: {
+			id: string;
+			delete?: boolean;
+			data: string[];
+		}[];
+		presence: {
+			[key: string]: {
+				setPresenceData?: () => void;
+			};
+		};
+	} = {
+		presence: {},
+		startedSince: Math.floor(Date.now() / 1000),
+	};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		type: ActivityType.Watching,
+		smallImageKey: Assets.Search,
+		smallImageText: "Browsing",
+		largeImageKey: Assets.Logo,
+		details: "Browsing...",
+		startTimestamp: data.startedSince,
+	};
+
+	data.presence = {
+		"/movie/([a-zA-Z0-9-]+)": {
+			setPresenceData() {
+				if (presence.isWatching()) {
+					const video = presence.getVideo();
+
+					[, presenceData.endTimestamp] =
+						presence.getTimestampsfromMedia(video);
+
+						if (presence.isTrailer()) {
+							presenceData.details = "Watching a trailer";
+							presenceData.state = presence.getVideoTitle().replace(/- (.+)/, "");
+
+							presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+							presenceData.smallImageKey = video.paused
+								? Assets.Pause
+								: Assets.Play;
+
+							presenceData.buttons = [
+								{
+									label: "Watch Trailer",
+									url: document.URL,
+								},
+							];
+						} else {
+							presenceData.details = `Watching ${presence.getVideoType() === "show" ? "show" : presence.getVideoType() === "movie" ? "movie" : "Unknown"}:`;
```

Comment:
Since `TOD.prototype.getVideoType` returns `"show" | "movie"`, these ternary statements may not be needed.
Perhaps something like this may work.
```ts
presenceData.details = `Watching a ${presence.getVideoType()}`;
```